system/security/LdapSecurity/ldapconnection.cpp

```diff
@@ -983,44 +983,25 @@ class CLdapConnection : implements ILdapConnection, public CInterface
         }
 
         int rc = LDAP_SERVER_DOWN;//assume bad things
-        StringBuffer hostbuf;
         for (int numHosts=0; numHosts < m_ldapconfig->getHostCount(); numHosts++)
         {
-            m_ldapconfig->getLdapHost(hostbuf);
-
+            StringBuffer ldapHost;
             for(int retries = 0; retries <= LDAPSEC_MAX_RETRIES; retries++)
             {
-                rc = connect(hostbuf.str(), proto);
-                if(!LdapServerDown(rc) || retries > LDAPSEC_MAX_RETRIES)
+                m_ldapconfig->getLdapHost(ldapHost);//get currently selected host, which may change
+                if(retries)
+                    DBGLOG("LDAP Server unreachable, retry attempt %d of %d on %s...", retries, LDAPSEC_MAX_RETRIES, ldapHost.str());
+                rc = connect(ldapHost.str(), proto);
+                if(rc == LDAP_SUCCESS)
+                    return true;
+                else if(rc == LDAP_SERVER_DOWN)
                     break;
                 sleep(LDAPSEC_RETRY_WAIT);
```

Comment:
There is no benefit in sleeping if the number of retries has expired.  Previously it would have exited the loop before sleeping.